modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts

```diff
@@ -748,14 +748,19 @@ export async function verifyWalletSignature(params: {
   commonKeychain: string;
   userKeyId: string;
   backupKeyId: string;
-  publicShare: string;
+  decryptedShare: string;
   verifierIndex: 1 | 2;
 }): Promise<void> {
   const rawNotations = await commonVerifyWalletSignature(params);
-  const publicShareRawNotationIndex = 2 + params.verifierIndex;
+  const publicUValueRawNotationIndex = 2 + params.verifierIndex;
 
+  // Derive public form of u-value
+  const publicUValue = ecc.pointFromScalar(Buffer.from(params.decryptedShare.slice(0, 64), 'hex'), false);
+  // Verify that the u value + chaincode is equal to the proof retrieved from the raw notations
   assert(
-    params.publicShare === Buffer.from(rawNotations[publicShareRawNotationIndex].value).toString(),
+    publicUValue !== null &&
```

Comment:
one thing missing in this function is to check the gpg pubkey Ids in the notation data.     in `verifyWalletSignatures()` you verified that the wallet sigs contain two distinct pgp keychains.      and you checked that the pubkeys themselves match the two we are interested in.

but in each keychain ([0] and [1] in verifyWalletSignatures()), there is notation data of not just the pub u-value and commonkeychain, but also the two gpg key Ids used as well.    

add a check to make sure those match up.

Reply:
This should actually be covered under `commonVerifyWalletSignature()`, which checks the first 3 notation data:
https://github.com/BitGo/BitGoJS/blob/master/modules/sdk-core/src/bitgo/tss/common.ts#L125-L150

Reply:
oh, perfect.